### PR TITLE
[🃏] NT-947 UpdateCardViewHolder

### PR DESCRIPTION
**Pull request by eoji**
If you want to check out the native project updates list, run the `NT-948` branch!

# 📲 What
Added new `ViewHolder` for Project Update cards.

# 🤔 Why
I'm breaking up my `NT-948` branch into digestable chunks.

# 🛠 How
- Added `isPublic` field to `Update` model. https://github.com/kickstarter/kickstarter/pull/19080
- Added `UpdateCardViewHolder` for Update cards with layout `item_update_card`.
  - Added `ic_chat_bubble` icon resource.
  - Added `celebrate_500` color resource
- Added `UpdateCardViewHolderViewModel` with all relevant outputs for Update cards.
- Added `UpdateCardViewHolderViewModelTest` to test all the outputs.

# 👀 See
This PR doesn't affect the UI! BTS only.

# 📋 QA
Does the code look good? 🤔

# Story 📖
[NT-947]

[NT-947]: https://kickstarter.atlassian.net/browse/NT-947

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.models.Project
+import com.kickstarter.models.Update
+import com.kickstarter.viewmodels.UpdateCardViewHolderViewModel
+import kotlinx.android.synthetic.main.item_update_card.view.*
+
+class UpdateCardViewHolder(private val view: View, val delegate: Delegate?) : KSViewHolder(view) {
+
+    interface Delegate {
+        fun updateClicked(update: Update)
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt]: `updateCardClicked`

File: app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                .compose(bindToLifecycle())
+                .compose(observeForUI())
+                .subscribe { this.view.update_sequence.text = this.ksString.format(updateSequenceTemplate, "update_count", it.toString()) }
+
+        this.viewModel.outputs.title()
+                .compose(bindToLifecycle())
+                .compose(observeForUI())
+                .subscribe { this.view.update_title.text = it }
+
+        this.viewModel.outputs.viewUpdate()
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt]: `showUpdateDetails`

File: app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    private fun setLikesCount(likesCount: Int) {
+        this.view.update_likes_count.text = likesCount.toString()
+        this.view.update_likes_count.contentDescription = this.ksString.format("likes_count_likes",
+                likesCount,
+                "likes_count",
+                NumberUtils.format(likesCount))
+    }
+
+    private fun setUpdatePublicUI(backersOnly: Boolean) {
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/ui/viewholders/UpdateCardViewHolder.kt]: `private fun setBackersOnlyVisibility(show: Boolean)`

File: app/src/main/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+import com.kickstarter.ui.viewholders.UpdateCardViewHolder
+import org.joda.time.DateTime
+import rx.Observable
+import rx.subjects.BehaviorSubject
+import rx.subjects.PublishSubject
+
+interface UpdateCardViewHolderViewModel {
+    interface Inputs {
+        /** Configure with the current [Project] and [Update]. */
+        fun configureWith(project: Project, update:Update)
```
Comment (jgsamudio) [inline app/src/main/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModel.kt]:
```suggestion
        fun configureWith(project: Project, update: Update)
```

File: app/src/test/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.libs.Environment
+import com.kickstarter.mock.factories.ProjectFactory
+import com.kickstarter.mock.factories.UpdateFactory
+import com.kickstarter.models.Update
+import org.joda.time.DateTime
+import org.junit.Test
+import rx.observers.TestSubscriber
+
+
```
Comment (eoji) [inline app/src/test/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModelTest.kt]: nl

File: app/src/test/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModelTest.kt [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+                .toBuilder()
+                .commentsCount(33)
+                .build()
+        this.vm.inputs.configureWith(ProjectFactory.project(), update)
+
+        this.commentsCountIsGone.assertValue(false)
+    }
+
+    @Test
+    fun testDate() {
```
Comment (eoji) [inline app/src/test/java/com/kickstarter/viewmodels/UpdateCardViewHolderViewModelTest.kt]: `testPublishDate`

**Pull request merged**